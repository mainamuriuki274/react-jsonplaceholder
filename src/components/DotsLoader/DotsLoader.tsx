import { FunctionComponent } from 'react';
import styles from './styles.module.scss';

const DotsLoader: FunctionComponent = () => (
  <div className={styles.loading_dots}>
    <div>.</div>
    <div>.</div>
    <div>.</div>
  </div>
);

export default DotsLoader;
