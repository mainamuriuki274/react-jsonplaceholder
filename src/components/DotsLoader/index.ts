export { default as DotsLoader } from './DotsLoader';
