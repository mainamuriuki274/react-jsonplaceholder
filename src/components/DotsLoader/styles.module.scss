.loading_dots > div {
  display: inline;
  font-size: 3.5em;
  opacity: 0;
  animation: showHideDot 1s ease-in-out infinite;
  &:nth-child(1) {
    animation-delay: 0.1s;
  }
  &:nth-child(2) {
    animation-delay: 0.3s;
  }
  &:nth-child(3) {
    animation-delay: 0.5s;
  }
}

@keyframes showHideDot {
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  60% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
