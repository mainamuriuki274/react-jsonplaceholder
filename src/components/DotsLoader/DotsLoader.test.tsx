import renderer from 'react-test-renderer';
import DotsLoader from './DotsLoader';

describe('Dots Loader Component', () => {
  it('it matches snapshot', () => {
    const tree = renderer.create(<DotsLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
