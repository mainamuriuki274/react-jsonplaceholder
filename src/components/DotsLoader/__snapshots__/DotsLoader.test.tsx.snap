// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dots Loader Component it matches snapshot 1`] = `
<div
  className="loading_dots"
>
  <div>
    .
  </div>
  <div>
    .
  </div>
  <div>
    .
  </div>
</div>
`;
